From ff1e30eeb9de43be89545da1fedd2aa71b3385de Mon Sep 17 00:00:00 2001
From: =?utf-8?q?Tommi=20My=C3=B6h=C3=A4nen?= <ext-tommi.1.myohanen@nokia.com>
Date: Wed, 26 Aug 2009 15:31:30 +0300
Subject: [PATCH] camerabin: fix viewfinder freezing in video recording

Camerabin changed it state to PAUSED and back to PLAYING when video
capturing was stopped. However, it caused random viewfinder freezes.
It looks like those state changes weren't necessary.
---
 gst/camerabin/gstcamerabin.c |    8 --------
 1 files changed, 0 insertions(+), 8 deletions(-)

diff --git a/gst/camerabin/gstcamerabin.c b/gst/camerabin/gstcamerabin.c
index 55efda2..4c51c0f 100644
--- a/gst/camerabin/gstcamerabin.c
+++ b/gst/camerabin/gstcamerabin.c
@@ -1950,14 +1950,6 @@ gst_camerabin_reset_to_view_finder (GstCameraBin * camera)
   if (camera->active_bin == camera->vidbin) {
     state_ret = gst_element_set_state (camera->active_bin, GST_STATE_READY);
 
-    /* Set to PAUSED to update new clock after stopping video recording */
-    /* Should we set videobin to NULL and post clock-lost message and 
-       let application do this? */
-    if (GST_STATE (camera) == GST_STATE_PLAYING) {
-      gst_element_set_state (GST_ELEMENT (camera), GST_STATE_PAUSED);
-      gst_element_set_state (GST_ELEMENT (camera), GST_STATE_PLAYING);
-    }
-
     if (state_ret == GST_STATE_CHANGE_FAILURE) {
       GST_WARNING_OBJECT (camera, "state change failed");
       gst_element_set_state (camera->active_bin, GST_STATE_NULL);
